operations/fetch.go

```diff
@@ -205,6 +205,9 @@ type cloneOptions struct {
 
 func clone(opts cloneOptions) error {
 	// clone the repo first
+	if opts.token == "" {
```

Comment:
It also seems like this is something that could be validated earlier on in the CLI as a before func, so if they run `evergreen fetch --source`, they also have to set a `--token`.